### PR TITLE
rds: add config reload time stat for rds

### PR DESCRIPTION
**Pull request by ramaraochavali**
We would like to know when a route update is loaded to Envoy. This PR adds a stat to indicate the same.
Commit Message: add config reload time stat.
Additional Description:
Risk Level: Low
Testing: Added
Docs Changes: Updated
Release Notes: Updated
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/configuration/overview/mgmt_server.rst [hunk 1]
```diff
@@ -53,6 +53,7 @@ The following statistics are generated for all subscriptions.
  :widths: 1, 1, 2
 
  config_reload, Counter, Total API fetches that resulted in a config reload due to a different config
+ config_reload_time, Gauge, Timestamp of the last config reload as milliseconds since the epoch
```
Comment (ramaraochavali) [inline docs/root/configuration/overview/mgmt_server.rst:56]: config_reload is only applicable/implemented for RDS/SRDS and VHDS - but added here generically. I followed the same pattern and added this new stat here even though it is implemented only for RDS and SRDS. If it is better to move to rds stats page - I can move.

File: test/common/router/rds_impl_test.cc [hunk 2]
```diff
@@ -290,6 +290,10 @@ TEST_F(RdsImplTest, Basic) {
   // Old config use count should be 1 now.
   EXPECT_EQ(1, config.use_count());
   EXPECT_EQ(2UL, scope_.counter("foo.rds.foo_route_config.config_reload").value());
+  EXPECT_NE(0, scope_
```
Comment (jmarantz) [inline test/common/router/rds_impl_test.cc:293]: could this flake if the test runs in <1ms?

If you are just trying to ensure that the gauge exists, you can use findGaugeByString, which is on class TestStore, which is the base class of MockIsolatedStatsStore.

Comment (ramaraochavali) [reply test/common/router/rds_impl_test.cc:293]: Sure.. Will change to ensure gauge exists.

**Comment by ramaraochavali** [comment]
@jmarantz PTAL addressed comments. The coverage failure not seems relevant.

File: source/common/router/rds_impl.h [hunk 3]
```diff
@@ -97,15 +97,16 @@ class StaticRouteConfigProviderImpl : public RouteConfigProvider {
 /**
  * All RDS stats. @see stats_macros.h
  */
-#define ALL_RDS_STATS(COUNTER)                                                                     \
+#define ALL_RDS_STATS(COUNTER, GAUGE)                                                              \
   COUNTER(config_reload)                                                                           \
+  GAUGE(config_reload_time_ms, NeverImport)                                                        \
```
Comment (jmarantz) [inline source/common/router/rds_impl.h:102]: nit: sort this block alphabetically, so the 2 counters go together.

File: source/common/router/scoped_rds.h [hunk 4]
```diff
@@ -84,6 +84,7 @@ class InlineScopedRoutesConfigProvider : public Envoy::Config::ImmutableConfigPr
 // clang-format off
 #define ALL_SCOPED_RDS_STATS(COUNTER, GAUGE)                                                       \
   COUNTER(config_reload)                                                                           \
+  GAUGE(config_reload_time_ms, NeverImport)                                                        \
```
Comment (jmarantz) [inline source/common/router/scoped_rds.h:87]: ditto

**Review by jmarantz — APPROVED** [review]
Looks great modulo nits.

/assign-from @envoyproxy/senior-maintainers

**Review by ggreenway — APPROVED** [review]

**Pull request merged**